### PR TITLE
Suggestions in the Scala book (3)

### PR DESCRIPTION
**Pull request by julienrf**
I’ve reviewed the chapters Methods, Functions, and Packaging and Imports.

I kindly request a review by @alvinj :blush:

### CODE REVIEW

File: _overviews/scala3-book/fun-hofs.md [hunk 1]
```diff
@@ -255,40 +255,40 @@ executeAndPrint(multiply, 3, 9)   // prints 27
 
 ## Function type signature consistency
 
-A great thing about learning about Scala’s function type signatures is that the syntax you use to define function input parameters is the same syntax you use to write anonymous functions and function variables.
+A great thing about learning about Scala’s function type signatures is that the syntax you use to define function input parameters is the same syntax you use to write functions literals.
```
Comment (alvinj) [inline _overviews/scala3-book/fun-hofs.md:258]: Change “functions” to “function” (without the ‘s’ at the end).

File: _overviews/scala3-book/methods-intro.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ next-page: methods-most
 
 
 In Scala 2, _methods_ can be defined inside classes, traits, objects, case classes, and case objects.
-But it gets better: In Scala 3 they can also be defined _outside_ any of those constructs with a new feature named Toplevel definitions.
+But it gets better: In Scala 3 they can also be defined _outside_ any of those constructs (we say that they are "top-level" definitions since they are not nested in another definition).
```
Comment (alvinj) [inline _overviews/scala3-book/methods-intro.md:12]: I would just add a comma after “definitions”.

**Comment by alvinj** [comment]
@julienrf, when I review these, is it better to click the Comment button when making a comment, or to press the Review button? I used the Comment approach with the “functions” suggestion, and Review button with the “comma after ‘definition’” suggestion.

File: _overviews/scala3-book/methods-most.md [hunk 3]
```diff
@@ -233,7 +234,7 @@ Here’s a method named `isTruthy` that implements the Perl definitions of `true
 
 ```scala
 def isTruthy(a: Any) =
-  if a == 0 || a == "" then
+  if a == 0 || a == "" || a == false then
```
Comment (alvinj) [inline _overviews/scala3-book/methods-most.md:237]: Wow, thanks, this is a great catch! No telling how many times I’ve written that wrong.

File: _overviews/scala3-book/methods-most.md [hunk 4]
```diff
@@ -279,7 +280,7 @@ d.speak()   // prints "Woof"
 ```
 
 Methods can also be marked as `private`.
-This makes them private to the current class, and they can’t be overridden in subclasses:
+This makes them private to the current class, they can’t be called nor overridden in subclasses:
```
Comment (alvinj) [inline _overviews/scala3-book/methods-most.md:283]: Change “they can’t” to “so they can’t”.

File: _overviews/scala3-book/methods-main-methods.md [hunk 5]
```diff
@@ -81,7 +81,7 @@ Illegal command line: java.lang.NumberFormatException: For input string: "sixty"
 The Scala compiler generates a program from an `@main` method `f` as follows:
 
 - It creates a class named `f` in the package where the `@main` method was found.
-- The class has a static method `main` with the usual signature: It takes an `Array[String]` as argument and returns `Unit`.
+- The class has a static method `main` with the usual signature of Java main methods: it takes an `Array[String]` as argument and returns `Unit`.
```
Comment (alvinj) [inline _overviews/scala3-book/methods-main-methods.md:84]: “Java main methods” could be changed to “Java `main` methods”.

File: _overviews/scala3-book/methods-intro.md (hunk 2 above)
Comment (alvinj) [inline _overviews/scala3-book/methods-intro.md:12]: If you want, you can include everything in the parentheses after a semi-colon, such as, “any of those constructs; we say that...”

I’m not sure how to articulate what I’m thinking, but I generally use the semi-colon approach, but it’s also fine either way.

**Review by alvinj — CHANGES_REQUESTED** [review]
These changes look great — thanks!

File: _overviews/scala3-book/fun-hofs.md (hunk 1 above)
Comment (julienrf) [reply _overviews/scala3-book/fun-hofs.md:258]: Fixed

File: _overviews/scala3-book/methods-intro.md (hunk 2 above)
Comment (julienrf) [reply _overviews/scala3-book/methods-intro.md:12]: Fixed.

File: _overviews/scala3-book/methods-most.md (hunk 4 above)
Comment (julienrf) [reply _overviews/scala3-book/methods-most.md:283]: Fixed

File: _overviews/scala3-book/methods-main-methods.md (hunk 5 above)
Comment (julienrf) [reply _overviews/scala3-book/methods-main-methods.md:84]: Fixed

File: _overviews/scala3-book/methods-intro.md (hunk 2 above)
Comment (julienrf) [reply _overviews/scala3-book/methods-intro.md:12]: Changed for a semi-colon :)

**Comment by julienrf** [comment]
> when I review these, is it better to click the Comment button when making a comment, or to press the Review button?

I like the review button because it sends me just one email notification :blush: 

Thank you for the review, Alvin!

**Comment by alvinj** [comment]
You’re welcome, I’m glad to help!

**Pull request merged**